### PR TITLE
WT-9765 Fix handling a prepared commit after a prepared rollback with eviction failure

### PR DESCRIPTION
**Pull request by quchenhao**

### CODE REVIEW

File: src/txn/txn.c
Comment (kommiharibabu) [inline src/txn/txn.c:1238]: Can you add a simple scenario explaining it would be much easier to understand?

File: test/suite/test_prepare25.py
Comment (kommiharibabu) [inline test/suite/test_prepare25.py:85]: The cursor is not closed here. It pins the page that avoids eviction. Close the cursor.

Comment (quchenhao) [reply test/suite/test_prepare25.py:85]: done

File: src/txn/txn.c
Comment (quchenhao) [reply src/txn/txn.c:1244]: done

**Review by kommiharibabu — APPROVED** [review]
LGTM

File: src/txn/txn.c
Comment (Siddhartha8899) [inline src/txn/txn.c:1235]: Shall we make it clear that this comment provides the reasoning for the `WT_UPDATE_TO_DELETE_FROM_HS ` flag check? 

Also, shouldn't the comment say:
` If the first committed update **before** the prepared update... `

**Review by Siddhartha8899 — CHANGES_REQUESTED** [review]

File: src/txn/txn.c
Comment (Siddhartha8899) [inline src/txn/txn.c:1236]: same here:
`we are in the case that there was a **next** prepared update . . .`

**Review by Siddhartha8899 — CHANGES_REQUESTED** [review]

File: test/suite/test_prepare25.py
Comment (Siddhartha8899) [inline test/suite/test_prepare25.py:51]: Just curious if this test fails without your changes.

File: src/txn/txn.c
Comment (quchenhao) [reply src/txn/txn.c:1235]: It should be first committed update older than the prepared upate

File: src/txn/txn.c
Comment (quchenhao) [reply src/txn/txn.c:1236]: I don't understand what you mean by next?

File: src/txn/txn.c
Comment (Siddhartha8899) [reply src/txn/txn.c]: Yup, that is better. Thanks for the update.

File: src/txn/txn.c
Comment (Siddhartha8899) [reply src/txn/txn.c]: I will resolve this as the updated comment makes sense.

**Review by Siddhartha8899 — APPROVED** [review]

**Comment by agorrod** [comment]
@quchenhao I forget what this is up to - is it OK to merge?

**Comment by quchenhao** [comment]
@agorrod Yes, I'll merge this after WT-9705 is merged.

**Comment by quchenhao** [comment]
evergreen merge

**Pull request merged**
